src/gateways/wayfinder.ts

```diff
@@ -8,24 +8,31 @@ import { EventType, trackEvent } from "~utils/analytics";
 export async function findGateway(
   requirements: Requirements
 ): Promise<Gateway> {
-  // get if the feature is enabled
+  // Get if the Wayfinder feature is enabled:
   const wayfinderEnabled = await getSetting("wayfinder").getValue();
 
-  // wayfinder disabled, but arns is needed
-  if (!wayfinderEnabled && requirements.arns) {
-    return {
-      host: "arweave.dev",
-      port: 443,
-      protocol: "https"
-    };
-  }
+  // This should have been loaded into the cache by handleGatewayUpdateAlarm, but sometimes this function might run
+  // before that, so in that case we fall back to the same behavior has having the Wayfinder disabled:
+  const procData = await getGatewayCache();
```

Comment:
Before, `procData` was `undefined` because the alarm handler that had to update the gateway cache would throw an error. I've added a check in case that happens again for whatever reason, but the original issues are fixed.